src/sage/rings/polynomial/multi_polynomial_element.py

```diff
@@ -333,9 +385,9 @@ def _div_(self, right):
             sage: x/S(2)
             1/2*x
         """
-        if right in self.base_ring():
-            inv = self.base_ring().one()/self.base_ring()(right)
-            return inv*self
+        if right.is_constant():
+            inv = self.base_ring().one() / self.base_ring()(right)
```

Comment:
```suggestion
            inv = self.base_ring().one() / right.constant_coefficient()
```
This is faster:
```python
sage: R.<x,y> = PolynomialRing(CyclotomicField(5)['q'].fraction_field())
sage: p = R(5)
sage: K = R.base_ring()
sage: %timeit p.constant_coefficient()
774 ns ± 30 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
sage: %timeit K(p)
2.14 µs ± 168 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```
Although I am not sure if `~right.constant_coefficient()` is better.

Reply:
I only changed a space here. I would prefer not touch anything.

Note that the division `polynomial / polynomial` is not complient with the coercion system. To my mind we should just remove `_div_` completely on polynomials and let the coercion framework works for us. But I would prefer postponing this discussion to another issue if that is ok with you. The fraction fields of polynomial rings needs more care than just adjusting this division function.

Reply:
You changed the line just prior and the change produces the exact same output faster (just like your change).

Reply:
Everything happening here is in the base ring.

Reply:
I changed as you suggested.